src/components/form/form.component.tsx

```diff
@@ -0,0 +1,88 @@
+import React from "react"
+
+import {FieldValues, useForm} from "react-hook-form"
+
+type FormFields = {
+	name: string
+	type: string
+	required?: string
+	pattern?: {
+		value: RegExp
+		message: string
+	}
+	minLength?: {
+		value: number
+		message: string
+	}
+}
+
+type FormProps = {
+	fields: FormFields[]
+	buttonText: string
+	onSubmitHandler: (data: FieldValues) => {}
+}
+
+const Form = ({fields, buttonText, onSubmitHandler}: FormProps) => {
+	const {
+		register,
+		handleSubmit,
+		formState: {errors, isSubmitting},
+	} = useForm()
+
+	return (
+		<div className="box-border flex flex-col justify-center items-center w-full">
+			<form
+				className="flex flex-col w-4/5 justify-center items-center sm:w-full sm:ml-0"
+				onSubmit={handleSubmit(async (data) => {
+					await onSubmitHandler(data)
+				})}
+			>
+				{fields.map((field: FormFields, index: number) => {
```

Comment:
You do not need to add types for field and index here. The map function already knows that it will return an item from the fields array of type FormFields. 
Also, index is not used anywhere so you can remove it from here.
```suggestion
				{fields.map((field) => {
```

Reply:
Understood n fixed.